File: agent/orchestrator.py
from schemas.agent import ChatRequest, ChatResponse
from agent.claude_sdk_runtime import ClaudeSdkAgentRuntime
from agent.model_adapter import ModelClient, get_model_client
from tools.registry import ToolRegistry, get_tool_registry


class AgentOrchestrator:
    def __init__(self, model_client: ModelClient, registry: ToolRegistry) -> None:
        self.runtime = ClaudeSdkAgentRuntime(model_client=model_client, registry=registry)

    async def chat(self, request: ChatRequest) -> ChatResponse:
        return await self.runtime.run(request)


def get_orchestrator() -> AgentOrchestrator:
    return AgentOrchestrator(
        model_client=get_model_client(),
        registry=get_tool_registry(),
    )

File: src/agent/orchestrator.ts
import type { ChatRequest, ChatResponse } from "../schemas/agent.js";
import { ClaudeSdkAgentRuntime } from "./claudeSdkRuntime.js";
import type { ModelClient } from "./modelAdapter.js";
import type { ToolRegistry } from "../tools/registry.js";

export class AgentOrchestrator {
  private readonly runtime: ClaudeSdkAgentRuntime;

  constructor(modelClient: ModelClient, registry: ToolRegistry) {
    this.runtime = new ClaudeSdkAgentRuntime(modelClient, registry);
  }

  async chat(request: ChatRequest): Promise<ChatResponse> {
    return this.runtime.run(request);
  }
}
